### PR TITLE
fix: do not refresh draft txn after redirect

### PR DESCRIPTION
**Pull request by dominictb**
### Details

avoid txn in Onyx get reset after IOU confirmation page refresh

### Fixed Issues

$ https://github.com/Expensify/App/issues/40936
PROPOSAL:  https://github.com/Expensify/App/issues/40936#issuecomment-2138147750

### Tests

1. Go to the app
2. Tap fab -- submit expense - scan
3. Upload an image using camera
4. Select a user
5. Enter description and save it
6. Refresh the page

Result: On refreshing scan submit expense confirmation page all details must not be disappeared

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Go to the app
2. Tap fab -- submit expense - scan
3. Upload an image using camera
4. Select a user
5. Enter description and save it
6. Refresh the page

Result: On refreshing scan submit expense confirmation page all details must not be disappeared

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/165644294/53c7afad-6914-435e-836e-199db3e29490

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/165644294/cb000d48-dd5f-41b1-b5d1-ff933a09c66f

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/165644294/b9d2d925-b12d-43e6-b25c-c908766d4c96

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/libs/actions/IOU.ts [hunk 1]
```diff
@@ -364,6 +364,20 @@ function initMoneyRequest(reportID: string, policy: OnyxEntry<OnyxTypes.Policy>,
     });
 }
 
+/**
+ *
+ * @param transactionID
+ * @param reportID
+ */
```
Comment (s77rt) [inline src/libs/actions/IOU.ts:371]: Missing isFromGlobalCreate

File: src/libs/actions/IOU.ts [hunk 2]
```diff
@@ -364,6 +364,20 @@ function initMoneyRequest(reportID: string, policy: OnyxEntry<OnyxTypes.Policy>,
...
+function resetMoneyRequest(transactionID: string, reportID: string, isFromGlobalCreate: boolean) {
```
Comment (s77rt) [inline src/libs/actions/IOU.ts:372]: This function does not reset the money request, only the participants and the report id. Can you please find a better fitting name?

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


android

</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/Expensify/App/assets/16493223/fe17d5b8-f9c7-448f-8b59-0c34f078a76b


</details>

<details>
<summary>iOS: Native</summary>


ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/Expensify/App/assets/16493223/1398af91-fba8-4652-8a1b-f1a2a2dfb902


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/Expensify/App/assets/16493223/b17cd224-13e1-4b4b-b8b0-7f89a4e233f8


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/16493223/fc329466-589a-4820-946c-aa62b0216c1f


</details>

File: src/pages/iou/request/IOURequestStartPage.tsx [hunk 3]
```diff
@@ -96,6 +96,12 @@ function IOURequestStartPage({
         if (transaction?.reportID === reportID) {
             return;
         }
+        // if transaction is from global create, the original behavior is that the user can change the recipient reportID
+        // without having to reset the whole transaction, hence we don't need to re-init the transaction in this case
+        if (transaction?.isFromGlobalCreate && transaction.transactionID) {
+            IOU.resetMoneyRequestReportID(transaction.transactionID, reportID, isFromGlobalCreate);
+            return;
+        }
```
Comment (s77rt) [inline src/pages/iou/request/IOURequestStartPage.tsx:104]: We don't need this anymore. Now the back button uses the transaction's report id

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/pages/iou/request/IOURequestStartPage.tsx (hunk 3 above)
Comment (s77rt) [reply src/pages/iou/request/IOURequestStartPage.tsx:104]: When navigating back to the participants page we will use the transaction's report id in the uri param. Thus the condition above `transaction?.reportID === reportID` will be met and we won't init the money request.

Comment (s77rt) [reply src/pages/iou/request/IOURequestStartPage.tsx:104]: In `navigateToStartStepIfScanFileCannotBeRead` can we navigate to the scan step?

Comment (s77rt) [reply src/pages/iou/request/IOURequestStartPage.tsx:104]: Currently the user is being navigated to the start step (`ROUTES.MONEY_REQUEST_CREATE_TAB_SCAN`). There is another page where the user can only select the receipt (`ROUTES.MONEY_REQUEST_STEP_SCAN`). Can we navigate to that page?

File: src/pages/iou/request/step/IOURequestStepConfirmation.tsx [hunk 4]
```diff
@@ -201,25 +203,32 @@ function IOURequestStepConfirmation({
...
-    useEffect(() => {
-        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-        const isLocalFile = FileUtils.isLocalFile(receiptPath);
+    useFocusEffect(
+        useCallback(() => {
+            const backToParam = ROUTES.MONEY_REQUEST_STEP_CONFIRMATION.getRoute(action, iouType, transactionID, reportID);
+            if (!receiptPath && requestType === CONST.IOU.REQUEST_TYPE.SCAN) {
+                Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_SCAN.getRoute(CONST.IOU.ACTION.CREATE, iouType, transactionID, reportID, backToParam));
+                return;
+            }
```
Comment (s77rt) [inline src/pages/iou/request/step/IOURequestStepConfirmation.tsx:214]: Why is this change required or what bug is it fixing?

File: src/pages/iou/request/step/IOURequestStepParticipants.tsx [hunk 5]
```diff
@@ -81,12 +82,29 @@ function IOURequestStepParticipants({
...
+    useFocusEffect(
+        useCallback(() => {
+            if (IOUUtils.isMovingTransactionFromTrackExpense(action)) {
+                return;
+            }
+            const backToParam = ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(iouType, transactionID, reportID, action);
+            if (!receiptPath && iouRequestType === CONST.IOU.REQUEST_TYPE.SCAN) {
+                Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_SCAN.getRoute(CONST.IOU.ACTION.CREATE, iouType, transactionID, reportID, backToParam));
+                return;
+            }
```
Comment (s77rt) [inline src/pages/iou/request/step/IOURequestStepParticipants.tsx:94]: Same ^

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/pages/iou/request/step/IOURequestStepConfirmation.tsx (hunk 4 above)
Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx:214]: Ah I see. Nice catch. For that we should just change the `backTo` to the start page. We'd also need to pass `shouldEnforceFallback` as true in the receipt scan page.

Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx:214]: - After user refresh the page and is navigated to the scan page, clicking the back button at this point should go back to the start page. Other pages (e.g. participant selection page) does not make sense because we won't have a receipt.
- If you can make it work without `shouldEnforceFallback=true` that's totally fine. I just didn't get this working as the fallback route was not in stack

Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx:214]: Navigating to the start page should be the simplest solution. You just need to modify navigateToStartStepIfScanFileCannotBeRead.onFailure callback adding an early return:
```js
        if (requestType === CONST.IOU.REQUEST_TYPE.SCAN) {
            Navigation.navigate(
                ROUTES.MONEY_REQUEST_STEP_SCAN.getRoute(CONST.IOU.ACTION.CREATE, iouType, transactionID, reportID, ROUTES.MONEY_REQUEST_START.getRoute(iouType, requestType)),
            );
            return;
        }
```

Then you can revert most of the other changes e.g. use of `useFocusEffect` and the extra navigation

Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx:214]: The `backTo` param won't be used in that case because it won't exist in the callstack. For the back button though we want to enforce it with `shouldEnforceFallback=true`



https://github.com/Expensify/App/assets/16493223/af664838-a4f1-42a3-a62c-881bc7dc9bc1

Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx:214]: @dominictb I'm trying to avoid "chasing the user" from page to another and instead just make them land in the correct page. Maybe we are missing a simpler solution here?

> It's already navigated to the scan step. Just that they navigate back using route.params.reportID instead of transaction.reportID -> causing the the money request to be re-init. If we want to use the transaction.reportID to nav back in this case, we'll (accidentally, or intentionally) force user to re-select participants.

Perhaps fixing the user re-selection would be more straightforward?

Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx:214]: I have also tried the solution in this PR and I'm getting stuck in the scan page


https://github.com/Expensify/App/assets/16493223/75abb52d-9c34-450e-a8c8-75c6b3f704c6

Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx:214]: @dominictb Redirecting the user to the start sounds good to me then. Can you please implement that?

Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx]: @dominictb Unfortunately this generated too much change. Let's just overwrite the go back functionality in the scan page if the request type is scan. Testing this and looks we only need the following changes:

`IOU.navigateToStartStepIfScanFileCannotBeRead`
```diff
-        if (requestType === CONST.IOU.REQUEST_TYPE.MANUAL) {
+        if (requestType === CONST.IOU.REQUEST_TYPE.MANUAL || requestType === CONST.IOU.REQUEST_TYPE.SCAN) {
```

`IOURequestStartPage.resetIOUTypeIfChanged`
```diff
+            if (transaction?.iouRequestType === newIOUType) {
+                return;
+            }
```

`IOURequestStepConfirmation.navigateBack`
```diff
-            Navigation.goBack(ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(iouType, transactionID, reportID, undefined, action));
+            Navigation.goBack(ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(iouType, transactionID, transaction?.reportID || reportID, undefined, action));
```

`IOURequestStepScan.navigateBack`
```diff
+        if (transaction?.iouRequestType === CONST.IOU.REQUEST_TYPE.SCAN) {
+            Navigation.goBack(ROUTES.MONEY_REQUEST_START.getRoute(iouType, transaction?.iouRequestType), true);
+            return;
+        }
```

Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx]: That looks like a workaround and it's hard to reason about. `initMoneyRequest` should reinit all the fields

Comment (s77rt) [reply src/pages/iou/request/step/IOURequestStepConfirmation.tsx]: My bad, I didn't read the solution well, I thought we are comparing the request type with `scan` type. Comparing the request type with the previous one looks good to me. Let's do that and test it

File: src/libs/actions/IOU.ts [hunk 6]
```diff
@@ -321,15 +321,35 @@ function getReportPreviewAction(chatReportID: string, iouReportID: string): Onyx
...
     const created = currentDate || format(new Date(), 'yyyy-MM-dd');
+
+    // In case we should keep existing transaction data, we only need to update the reportID, isFromGlobalCreate, created and currency
+    if (shouldKeepExistingData) {
+        Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${newTransactionID}`, {
+            reportID,
+            isFromGlobalCreate,
+            created,
+            currency,
+            transactionID: newTransactionID,
```
Comment (s77rt) [inline src/libs/actions/IOU.ts:348]: Why are we resetting all these props? Shouldn't be just the report id?

File: src/libs/actions/IOU.ts [hunk 7]
```diff
@@ -321,15 +321,35 @@ function getReportPreviewAction(chatReportID: string, iouReportID: string): Onyx
...
+ * @param shouldKeepExistingData
  * @param iouRequestType one of manual/scan/distance
- * @param skipConfirmation if true, skip confirmation step
  */
-function initMoneyRequest(reportID: string, policy: OnyxEntry<OnyxTypes.Policy>, isFromGlobalCreate: boolean, iouRequestType: IOURequestType = CONST.IOU.REQUEST_TYPE.MANUAL) {
+function initMoneyRequest(
+    reportID: string,
+    policy: OnyxEntry<OnyxTypes.Policy>,
+    isFromGlobalCreate: boolean,
+    shouldKeepExistingData: boolean,
```
Comment (s77rt) [inline src/libs/actions/IOU.ts:331]: Let's not give the user the option to control this. It will create confusion. Instead we should determine this in the function itself

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/libs/actions/IOU.ts [hunk 8]
```diff
@@ -321,15 +321,33 @@ function getReportPreviewAction(chatReportID: string, iouReportID: string): Onyx
  * @param reportID to attach the transaction to
  * @param policy
  * @param isFromGlobalCreate
+ * @param shouldKeepExistingData
```
Comment (s77rt) [inline src/libs/actions/IOU.ts:324]:
```suggestion
```

**Review by s77rt — CHANGES_REQUESTED** [review]
Looks good just a NAB comment

**Comment by s77rt** [comment]
@dominictb Can you please add missing screenshots / videos?

**Review by s77rt — APPROVED** [review]

**Review by puneetlath — APPROVED** [review]
